platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/offline/OfflineManager.java

```diff
@@ -0,0 +1,172 @@
+package com.mapbox.mapboxsdk.offline;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Looper;
+import android.support.annotation.NonNull;
+
+import java.io.File;
+
+/**
+ * The offline manager is the main entry point for offline-related functionality.
+ * It'll help you list and create offline regions.
+ */
+public class OfflineManager {
+
+    // Default database name and size (1GB)
+    private final static String OFFLINE_DATABASE_NAME = "mbgl-offline.db";
+    private final static long OFFLINE_MAX_CACHE_SIZE = 1024 * 1024 * 1024;
```

Comment:
FYI: this limit is applied to _non-offline_ resources only, i.e. resources added to the database for the "ambient use" caching functionality. There is no size limit for offline resources. So 1GB is probably too high; if a user never creates any offline regions, we want the database to remain fairly small (order tens or low hundreds of megabytes).